src/core/qfieldcloudprojectsmodel.cpp

```diff
@@ -1809,6 +1809,26 @@ void QFieldCloudProjectsModel::downloadFileConnections( const QString &projectId
   } );
 
   connect( reply, &NetworkReply::downloadProgress, reply, [=]( int bytesReceived, int bytesTotal ) {
+    const QString temporaryFileName = project->downloadFileTransfers[fileName].tmpFile;
+    QNetworkReply *rawReply = reply->reply();
+    QFile file( temporaryFileName );
+
+    if ( file.open( QIODevice::WriteOnly | QIODevice::Append ) )
+    {
+      file.write( rawReply->readAll() );
+
+      if ( file.error() != QFile::NoError )
+      {
+        QgsLogger::debug( QStringLiteral( "Failed to write on temporary file `%1`." ).arg( temporaryFileName ) );
```

Comment:
I would keep as much details as possible about the failure, such as original filename, project id etc. 

Also note that before the failures were logged also to the user in `QgsMessageLog::logMessage`, I would prefer to have this back.